homework01/5testng/src/test/java/TypeCheckingTest.java

```
@@ -0,0 +1,93 @@
+import junit.framework.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+
+public class TypeCheckingTest {
+
+    @DataProvider(name = "isosceles")
+    public Object[][] getIsosceles() {
+        return new Object[][]{
+                {2, 2, 3},
+                {3, 2, 2},
+                {2, 3, 2},
+                {Double.MAX_VALUE, Double.MAX_VALUE,Double.MIN_VALUE},
+                {Double.MIN_VALUE, 1,1},
+        };
+    }
+
+    @DataProvider(name = "equilateral")
+    public Object[][] getEquilateral() {
+        return new Object[][]{
+                {1, 1, 1},
+                {Double.MAX_VALUE, Double.MAX_VALUE, Double.MAX_VALUE},
+                {Double.MIN_VALUE, Double.MIN_VALUE, Double.MIN_VALUE},
+                {Double.MIN_NORMAL, Double.MIN_NORMAL, Double.MIN_NORMAL},
+        };
+    }
+
+    @DataProvider(name = "normal")
+    public Object[][] getNormal() {
+        return new Object[][]{
+                {2, 2.2, 2.3},
+                {2.3, 2, 2.2},
+                {2.2, 2.3, 2},
+                {Double.MAX_VALUE/2, Double.MAX_VALUE/3, Double.MAX_VALUE/4},
```

It's kind of magic to hide an error. One of the argument has to be Double.MAX_VALUE